qemu/qmp.go

```diff
@@ -722,11 +722,16 @@ func (q *QMP) ExecuteQuit(ctx context.Context) error {
 	return q.executeCommand(ctx, "quit", nil, nil)
 }
 
-// ExecuteBlockdevAdd sends a blockdev-add to the QEMU instance.  device is the
-// path of the device to add, e.g., /dev/rdb0, and blockdevID is an identifier
-// used to name the device.  As this identifier will be passed directly to QMP,
-// it must obey QMP's naming rules, e,g., it must start with a letter.
-func (q *QMP) ExecuteBlockdevAdd(ctx context.Context, device, blockdevID string) error {
+// ExecuteBlockdevAddWithCache sends a blockdev-add to the QEMU instance.
+// device is the path of the device to add, e.g., /dev/rdb0, and blockdevID
+// is an identifier used to name the device.  As this identifier will be
+// passed directly to QMP, it must obey QMP's naming rules, e,g., it must
+// start with a letter.
+// direct and noflash are cache-related options for block devices that
+// are described in https://github.com/qemu/qemu/blob/master/qapi/block-core.json.
+// direct denotes if enables use of O_DIRECT (bypass the host page cache).
+// noflash denotes if ignore any flush requests for the device.
+func (q *QMP) ExecuteBlockdevAddWithCache(ctx context.Context, device, blockdevID string, direct, noflash bool) error {
```

Comment:
Should we also add support for writeback for completeness, even if it is not being used right now?

Reply:
I tried but "blockdev-add" doesn't support writeback.
This is the introduction of them in https://github.com/qemu/qemu/blob/master/qapi/block-core.json:
##
# @BlockdevCacheOptions:
#
# Includes cache-related options for block devices
#
# @direct:      enables use of O_DIRECT (bypass the host page cache;
#               default: false)
# @no-flush:    ignore any flush requests for the device (default:
#               false)
#
# Since: 2.9
##
{ 'struct': 'BlockdevCacheOptions',
  'data': { '*direct': 'bool',
            '*no-flush': 'bool' } }